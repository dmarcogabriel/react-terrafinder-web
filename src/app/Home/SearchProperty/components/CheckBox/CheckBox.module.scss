@import 'styles';

.container {
  display: flex;
  align-items: center;
  margin: 1rem 0;

  > p {
    margin-left: 1rem;
  }
}
