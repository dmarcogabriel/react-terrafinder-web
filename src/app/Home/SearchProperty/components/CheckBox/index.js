import React from 'react';
import { MdCheckBox, MdCheckBoxOutlineBlank } from 'react-icons/md';
import { container } from './CheckBox.module.scss';

const ICON_SIZE = 22;

export default function CheckBox({ label, selected, onChange }) {
  return (
    <button
      type="button"
      className={container}
      onClick={() => onChange(!selected)}
    >
      {selected ? (
        <MdCheckBox size={ICON_SIZE} />
      ) : (
        <MdCheckBoxOutlineBlank size={ICON_SIZE} />
      )}
      <p>{label}</p>
    </button>
  );
}
