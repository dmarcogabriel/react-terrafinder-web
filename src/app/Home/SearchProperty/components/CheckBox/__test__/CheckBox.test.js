import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import CheckBox from '..';

describe('<CheckBox />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CheckBox />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<CheckBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
