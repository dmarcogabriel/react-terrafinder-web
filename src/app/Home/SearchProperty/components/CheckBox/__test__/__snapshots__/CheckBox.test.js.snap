// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> matches snapshot 1`] = `
<button
  data-testid="checkbox"
  onClick={[Function]}
  type="button"
>
  <svg
    fill="currentColor"
    height={22}
    size={22}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={22}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
    />
  </svg>
  <p />
</button>
`;
